cli/commands/scaffold/action.go

```diff
@@ -0,0 +1,351 @@
+package scaffold
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/gruntwork-io/terragrunt/config"
+	"github.com/gruntwork-io/terragrunt/shell"
+
+	"github.com/gruntwork-io/terragrunt/terraform"
+
+	"github.com/gruntwork-io/terragrunt/cli/commands/hclfmt"
+	"github.com/gruntwork-io/terragrunt/util"
+
+	boilerplate_options "github.com/gruntwork-io/boilerplate/options"
+	"github.com/gruntwork-io/boilerplate/templates"
+	"github.com/gruntwork-io/boilerplate/variables"
+	"github.com/gruntwork-io/go-commons/errors"
+	"github.com/gruntwork-io/terragrunt/options"
+	"github.com/gruntwork-io/terratest/modules/files"
+	"github.com/hashicorp/go-getter"
+)
+
+const (
+	moduleCount            = 2
+	moduleAndTemplateCount = 3
+
+	sourceUrlTypeHttps = "git-https"
+	sourceUrlTypeGit   = "git-ssh"
+	sourceGitSshUser   = "git"
+
+	sourceUrlTypeVar    = "SourceUrlType"
+	sourceGitSshUserVar = "SourceGitSshUser"
+	refVar              = "Ref"
+	// refParam - ?ref param from url
+	refParam = "ref"
+
+	moduleUrlPattern = `git::([^:]+)://([^/]+)(/.*)`
+	moduleUrlParts   = 4
+
+	defaultBoilerplateConfig = `
```

Comment:
- I think the  more "idiomatic" way to use boilerplate is to have these as files on disk in a folder (i.e., `boilerplate.yml`, `terragrunt.hcl`, etc), and to run `boilerplate` against that folder.  
- This "default template" is just our starting point for the `scaffold` command. We'll want other built-in templates for Terragrunt, such as scaffolding out a new module (e.g., `main.tf`, `variables.tf`, etc), scaffolding out a wrapper module, scaffolding out an entire `infra-live` repo, and so on. So all of these templates should live in folders in a GitHub repo: e.g., a `boilerplate` folder in the Terragrunt repo, with this default template under `boilerplate/default-terragrunt-module`. 
- The `--template-url` arg of `boilerplate` already supports URLs, so you could point it to the TG repo URL (which is open source) and it should "just work."

That said, it's OK for this first version to leave the default template "embedded" in the binary like this. It also makes local testing easier (while iterating on the template), which is something we'd have to solve for with the approach above.

So, for now, could you file a GH issue for this, and we can come back to it a bit later?